server/src/test/java/org/elasticsearch/index/shard/RefreshListenersTests.java

```diff
@@ -97,6 +99,8 @@ public void setupListeners() throws Exception {
         maxListeners = randomIntBetween(2, 1000);
         // Now setup the InternalEngine which is much more complicated because we aren't mocking anything
         threadPool = new TestThreadPool(getTestName());
+        IndexSettings indexSettings = IndexSettingsModule.newIndexSettings("index", Settings.EMPTY);
+        threadPoolMergeQueue = ThreadPoolMergeQueue.getNewThreadPoolMergeQueue(threadPool, indexSettings.getNodeSettings());
```

Comment:
Is this not equivalent to:

```suggestion
        threadPoolMergeQueue = ThreadPoolMergeQueue.getNewThreadPoolMergeQueue(threadPool, Settings.EMPTY);
```
?

Perhaps we should randomize the setting though?

Reply:
Yes, pushed c3d40016.